changelog/objc_class.dd

```diff
@@ -0,0 +1,49 @@
+Add support for Objective-C classes
+
+Prior to this release D interfaces were used to represent Objective-C classes.
+Now proper support for Objective-C classes has been added and D classes can
+be used instead to represent Objective-C classes. It's preferred to use D
+classes to represent Objective-C classes.
+
+This release also adds support for implementing Objective-C subclasses and
+methods in D.
+
+To match the behavior in Objective-C some additional changes have been made:
+
+$(UL
+    $(LI `static` and `final` methods are virtual. Although `final` methods are
+         virtual it's not possible to override a `final` method in a subclass)
+    $(LI `static` methods are overridable in subclasses)
```

Comment:
That sounds quite fishy. An `extern` declaration is used for compatibility, but we always stayed away from importing other language's semantic into D, as it's a hornets nest. It's fine to have `final` method as virtual if ObjC has no way to represent it, but allowing `override static` is downright confusing.

I assume this comes from the necessity to have it in the vtable, even if its static ?

Reply:
Class/static methods in Objective-C or instance methods on the metaclass. In Objective-C classes are object themselves.

Reply:
> I assume this comes from the necessity to have it in the vtable, even if its static ?

There is no vtable for Objective-C classes. All method calls are lowered to a function in the Objective-C runtime.

Reply:
Without this, it is basically impossible to actually use existing Objective C classes. We sometimes need to be practical even when it sounds weird...

Reply:
@jacob-carlborg : Is there a way to expose this without changing the language ?

Reply:
Expose what? That static methods are overridable?